internal/controller/nimcache_controller.go

```diff
@@ -174,6 +176,16 @@ func (r *NIMCacheReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 			return ctrl.Result{}, nil
 		}
 	}
+
+	// Set container orchestrator type
+	if r.GetOrchestratorType() != "" {
```

Comment:
NIT: Can you just fold the `k8sutil.GetOrchestratorType(r.GetClient())` into `GetOrchestratorType()`? It will easier if we end up calling `GetOrchestratorType()` at other places

Reply:
yeah, good idea, whenever it is not set we can invoke the API call.

Reply:
updated